File: amazia-core/src/main/java/com/example/delivery/controller/GetShippingLeadTimeController.java
package com.example.delivery.controller;

import com.example.delivery.dto.ShippingLeadTimeResponse;
import com.example.delivery.service.GetShippingLeadTimeService;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

@RestController
@RequestMapping("/api")
public class GetShippingLeadTimeController {

    private final GetShippingLeadTimeService service;

    public GetShippingLeadTimeController(GetShippingLeadTimeService service) {
        this.service = service;
    }

    @GetMapping("/shipping-lead-times/{id}")
    public ResponseEntity<ShippingLeadTimeResponse> get(@PathVariable Long id) {
        return ResponseEntity.ok(service.get(id));
    }
}

File: amazia-console/app/Delivery/Controller/GetShippingLeadTimeController.php
<?php

namespace App\Delivery\Controller;

use App\Delivery\Service\GetShippingLeadTimeService;
use App\Http\Controllers\Controller;
use Illuminate\Http\Request;

/**
 * 都道府県別リードタイム詳細 Controller（GET /api/shipping-lead-times/{id}）。
 */
class GetShippingLeadTimeController extends Controller
{
    private GetShippingLeadTimeService $service;

    public function __construct(GetShippingLeadTimeService $service)
    {
        $this->service = $service;
    }

    public function __invoke(Request $request, int $id)
    {
        $role = (string) $request->get('auth_role');
        if (!in_array($role, config('app.auth.approver_roles'), true)) {
            return response()->json(['message' => 'Forbidden'], 403);
        }

        $response = $this->service->get($id);
        return response()->json($response->json(), $response->status());
    }
}
